### PR TITLE
[SPARK-2827][GraphX] Add collectDegreeDist to compute the distribute of vertex degrees in GraphOps

### PR DESCRIPTION
**Pull request by maropu**
Add degree distribution operators in GraphOps for GraphX.

### CODE REVIEW

**Comment by maropu** [comment]
Refactored #1767 and added unit tests.
I made the original patch simpler since this is a first patch to add a new API.
If necessary, the other part will be add in a following patch.

**Comment by ankurdave** [comment]
ok to test

**Comment by andrewor14** [comment]
ok to test

File: graphx/src/test/scala/org/apache/spark/graphx/util/FrequencyDistributionSuite.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  test("FrequencyDistribution.equiwidth") {
+    withSpark { sc =>
+      val rdd = sc.parallelize(Seq[Int](1, 2, 2, 1, 3, 1, 3, 4, 8, 1, 9))
+      val freq = FrequencyDistribution.equiwidth(2).compute(rdd)
+      assert(freq.toSeq == Seq(((1, 2), 6), ((3, 4), 3), ((5, 6), 0), ((7, 8), 1), ((9, 10), 1)))
+    }
+  }
+}
```
Comment (andrewor14) [inline graphx/src/test/scala/org/apache/spark/graphx/util/FrequencyDistributionSuite.scala]: need new line

**Comment by andrewor14** [comment]
@ankurdave @maropu what's the status of this patch? Is it awaiting reviews? Is it still active?

File: graphx/src/main/scala/org/apache/spark/graphx/GraphOps.scala [hunk 2]
```diff
@@ -186,6 +188,39 @@ class GraphOps[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED]) extends Seriali
...
+          (id, data, degree) => degree.getOrElse(0))
+      case EdgeDirection.In =>
+        graph.vertices.leftJoin(inDegrees)(
+          (id, data, degree) => degree.getOrElse(0))
+      case EdgeDirection.Out =>
+        graph.vertices.leftJoin(outDegrees)(
+          (id, data, degree) => degree.getOrElse(0))
+      case _ =>
+        throw new SparkException("collectDegreeDist does not support EdgeDirection.Both. Use" +
+          "EdgeDirection.Either instead.")
```
Comment (ankurdave) [inline graphx/src/main/scala/org/apache/spark/graphx/GraphOps.scala]: Need a space in the string concatenation

File: graphx/src/main/scala/org/apache/spark/graphx/GraphOps.scala [hunk 3]
```diff
@@ -186,6 +188,39 @@ class GraphOps[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED]) extends Seriali
...
+    val vertexDegrees = edgeDirection match {
+      case EdgeDirection.Either =>
+        graph.vertices.leftJoin(degrees)(
+          (id, data, degree) => degree.getOrElse(0))
+      case EdgeDirection.In =>
+        graph.vertices.leftJoin(inDegrees)(
+          (id, data, degree) => degree.getOrElse(0))
+      case EdgeDirection.Out =>
+        graph.vertices.leftJoin(outDegrees)(
+          (id, data, degree) => degree.getOrElse(0))
```
Comment (ankurdave) [inline graphx/src/main/scala/org/apache/spark/graphx/GraphOps.scala]: minor: these can fit on one line

**Comment by ankurdave** [comment]
LGTM aside from minor comments

**Comment by maropu** [comment]
Ok, I'll fix it.

**Comment by maropu** [comment]
@ankurdave @andrewor14 Fixed and could you merge this?

**Comment by maropu** [comment]
test this please

**Comment by andrewor14** [comment]
retest this please

**Comment by maropu** [comment]
@andrewor14 Why this test does not work?

**Comment by maropu** [comment]
retest this please

**Comment by andrewor14** [comment]
I'm not sure. retest this please?

**Comment by maropu** [comment]
retest this please

**Comment by andrewor14** [comment]
Weird, something is wrong with this PR. Maybe re-open it and see if the problem goes away?

**Comment by maropu** [comment]
@andrewor14 Okay

**Pull request closed without merging**